contracts/reserve-auction/src/execute.rs

```diff
@@ -79,113 +85,132 @@ pub fn execute_create_auction(
     env: Env,
     collection: Addr,
     token_id: &str,
-    start_time: Timestamp,
-    end_time: Timestamp,
+    duration: u64,
     reserve_price: Coin,
     seller_funds_recipient: Option<Addr>,
 ) -> Result<Response, ContractError> {
     let config = CONFIG.load(deps.storage)?;
 
-    only_owner(deps.as_ref(), &info, &collection, token_id)?;
-    has_approval(deps.as_ref(), &env.contract.address, &collection, token_id)?;
+    // Only NFT owner can create auction for an NFT
+    only_owner(&deps.querier, &info, &collection, token_id)?;
+
+    // NFT owner must have approved the reserve auction contract to transfer the NFT
+    has_approval(
+        &deps.querier,
+        &env.contract.address,
+        &collection,
+        token_id,
+        Some(false),
+    )?;
+
+    // Cannot create a duplicate auction for an NFT
     only_no_auction(deps.as_ref(), &collection, token_id)?;
 
+    only_tradable(&deps.querier, &env.block, &collection)?;
+
     let mut response = Response::new();
 
-    let fee = must_pay(&info, NATIVE_DENOM)?;
-    if fee != config.create_auction_fee {
-        return Err(ContractError::WrongFee {
-            required: config.create_auction_fee,
-            given: fee,
-        });
-    }
-    fair_burn(fee.u128(), None, &mut response);
+    validate_reserve_price(deps.as_ref().storage, &reserve_price)?;
 
-    if !has_coins(&[reserve_price.clone()], &config.min_reserve_price) {
-        return Err(ContractError::InvalidReservePrice {
-            min: config.min_reserve_price,
-        });
-    }
-
-    if start_time < env.block.time {
-        return Err(ContractError::InvalidStartTime {});
-    }
-    if end_time < start_time.plus_seconds(config.min_duration) {
-        return Err(ContractError::InvalidEndTime {});
+    // Ensure that the duration is within the min and max duration
+    ensure!(
+        duration >= config.min_duration && duration <= config.max_duration,
+        ContractError::InvalidDuration {
+            min: config.min_duration,
+            max: config.max_duration,
+            got: duration
+        }
+    );
+
+    // Handle create auction fee payment
+    if config.create_auction_fee.amount.is_zero() {
+        // If create auction fee is zero, ensure no payment is sent
+        nonpayable(&info)?;
+    } else {
+        // If create auction fee is non zero, ensure user has sent the correct amount,
+        // and send it to the fair-burn contract
+        ensure!(
+            has_coins(&info.funds, &config.create_auction_fee),
+            ContractError::WrongFee {
+                expected: config.create_auction_fee,
+            }
+        );
+        response = append_fair_burn_msg(
+            &config.fair_burn,
+            vec![config.create_auction_fee],
```

Comment:
```suggestion
            coins(config.create_auction_fee),
```

Reply:
hmm `create_auction_fee` is actually a `Coin` so it doesn't fit into coins which has the below signature:

`pub fn coins(amount: u128, denom: impl Into<String>)`